packages/ai-jsx/src/next/index.ts

```diff
@@ -0,0 +1,114 @@
+import 'server-only';
+import * as ReactModule from 'react';
+import * as LLMx from '../react/core.js';
+export * from '../react/core.js';
+import { markAsJsxBoundary } from '../react/jsx-boundary.js';
+
+function unwrapReact(partiallyRendered: LLMx.PartiallyRendered): ReactModule.ReactNode {
+  if (LLMx.isElement(partiallyRendered)) {
+    // This should be an AI.React element.
+    if (partiallyRendered.tag !== LLMx.React) {
+      throw new Error('unwrapReact only expects to see AI.React elements or strings.');
+    }
+
+    return partiallyRendered.props.children;
+  }
+
+  return partiallyRendered;
+}
+
+function computeSuffix(prefix: ReactModule.ReactNode[], frame: ReactModule.ReactNode[]) {
+  const computedSuffix = [] as ReactModule.ReactNode[];
+  for (let i = 0; i < frame.length; ++i) {
+    if (i >= prefix.length) {
```

Comment:
minor: it might be cleaner to concat the entire remainder of the string once `i` exceeds the prefix length: 

```
if (i >= prefix.length) {
    computedSuffix.push(...frame.slice(i))
}
```